services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/OpenHouseInternalRepositoryImpl.java

```diff
@@ -472,9 +473,25 @@ public boolean existsById(TableDtoPrimaryKey tableDtoPrimaryKey) {
   @Timed(metricKey = MetricsConstant.REPO_TABLE_DELETE_TIME)
   @Override
   public void deleteById(TableDtoPrimaryKey tableDtoPrimaryKey) {
+    Table table =
+        catalog.loadTable(
+            TableIdentifier.of(
+                tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()));
+    String location = table.location();
+    String tableCreator = table.properties().get("openhouse.tableCreator");
     catalog.dropTable(
         TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()),
-        true);
+        false);
+
+    try {
+      storageManager.getDefaultStorage().deallocateTableLocation(location, tableCreator);
+    } catch (IOException e) {
+      String errMsg =
+          String.format(
+              "Exception while deleting table location %s for %s.%s",
```

Comment:
Consider logging the table location. We can skip location details in errors thrown back to clients to maintain table abstraction.